input/fsh/profiles/group.fsh

```diff
@@ -0,0 +1,78 @@
+Title: "Members Refreshed"
+Description: "Extension used by a server to indicate to a client when the members in a dynamic group were last updated"
+Context: Group
+* value[x] only dateTime
+
+
+Profile: BulkCohortGroup
+Parent: Group 
+Id: bulk-cohort-group
+Title: "Bulk Cohort Group"
+Description: "Group that provides characteristic based cohorts through coarse-grained, REST search expression based filters to support constraining bulk export requests"
+* type from GroupTypeSubset (required)
+  * ^definition = """
+    A client SHALL populate this element with `person` when creating a group of Patients, or `practitioner` when creating a group of Practitioners.
+    """
+* member
+  * ^definition = """
+    A server MAY support the inclusion of one or more `member` elements that contain an `entity` element with a reference to a Patient resource, Practitioner resource, or Group resource that is a group of Patient resources or Practitioner resources. When members are provided, the expression in the `member-filter` extension for the Group SHALL only be applied to the referenced resources and the compartments of the referenced resources, or those of the members of referenced Group resources. When members are not provided and the Group's `type` element is set to `person`, the expression in the `member-filter` extension SHALL be applied to all of the Patient resources and Patient compartments the client is authorized to access. When members are not provided and the Group's `type` element is set to `practitioner`, the expression in the `member-filter` extension SHALL be applied to all of the Practitioner resources and Practitioner compartments the client is authorized to access.
+    """
+* modifierExtension contains MemberFilter named member-filter 1..*
+  * ^short = "Filter for members of this group" 
+  * ^definition = """
+    A server SHALL support the inclusion of one or more `member-filter` modifier extensions containing a `valueExpression` with a language of `application/x-fhir-query` and an `expression` populated with a FHIR REST API query for a Patient or Practitioner resource or a resource type included in the Patient or Practitioner compartment. If multiple `member-filter` extensions are provided that contain criteria for different resource types, servers SHALL filter the group to only include Patients or Practitioners whose resources and resources in their compartments meet the conditions in all of the expressions. If multiple `member-filter` extensions are provided that contain criteria for a single resource type, the server SHALL include Patients or Practitioners whose resources and resources in their compartments meet the criteria for that resource type in any of those expressions (a logical "or"). A server MAY also support other expression languages such as `text/cql`. When more than one language is supported by a server a client SHALL use a single language type for all of the member-filter expressions included in a single Group.
```

Comment:
This "and within a type, or across types" feels dodgy to me, seeing with fresh eyes. Like in our _typeFilter logic, if you repeat a filter for a given type, it's an "or", so clearly we think "or within a type" can be useful?

Reply:
Think this is maybe backwards - think we're proposing "and" between types and "or" within a type. The "or" behavior is consistent with _typeFilter, but the "and" behavior is different to make things like "18+ and diabetic" easy at the cost of "18+ or diabetic". Looking at it that way, does it seem like the right behavior to you or do you think it should always be "or"?